### PR TITLE
Add flash id component to IotFlashInfo struct

### PR DESCRIPTION
**Pull request by cmcgcc**
Description
-----------
Add flash id component to IotFlashInfo struct

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by cmcgcc** [comment]
Please help for code review. @vjjunnut @cobusve @brianfr @dongh @qiutongs @lundinc2

File: libraries/abstractions/common_io/include/iot_flash.h [hunk 1]
```diff
@@ -94,6 +94,7 @@ typedef struct IotFlashInfo
                                  *   and if sub-sectors of 4k can be write protected then this may be set to 4K*/
     uint8_t ucAsyncSupported;   /*!< Provides a flag to indicate if background erase/program
                                  *   is supported. User must check this before they can setup a callback and wait.*/
+    uint32_t ulFlashID;         /*!< flash chip id */
```
Comment (lundinc2) [inline libraries/abstractions/common_io/include/iot_flash.h:97]: Nit: Do you mind changing this to "Flash chip ID."

Comment (cmcgcc) [reply libraries/abstractions/common_io/include/iot_flash.h:97]: The rest flash components like ulFlashSize in the structure are commented without capital F. It is not shown in this diff.

Comment (lundinc2) [reply libraries/abstractions/common_io/include/iot_flash.h:97]: Okay, sounds good to me

**Pull request merged**